### DISCUSSION TITLE
How much certificate chain path length does strongswan support to form a tunnel?

### DISCUSSION THREAD

**Discussion by demahaj1** [strongswan/strongswan#942] [Q&A] [answered]
Recently I was trying to install chain of certificate with path length of 8 intermediate CA.. the certificates were installed but tunnel was not formed...

Is there any specific chain of trust file size does strongswan limits to form a tunnel ?
Do we have limited number of intermediate chain length supported, may be max 7 to form tunnel?

Can you please provide your inputs on this observation.. thanks

**Comment by strongX509** [comment, marked as answer]
In order to prevent endless looping with intermediate certificates referencing each other, the maximum chain length is hardcoded in 

 https://github.com/strongswan/strongswan/blob/master/src/libstrongswan/credentials/credential_manager.c#L32

to  **#define MAX_TRUST_PATH_LEN 7**

**Comment by demahaj1** [comment]
@strongX509 thanks for the response..
